### PR TITLE
DPC-1255: Replicate in implementer portal - login page

### PR DESCRIPTION
**Pull request by Sun-Mountain**
### Fixes [DPC-1255](https://jira.cms.gov/browse/DPC-1255)

Ensure user can log in, reset password, and the passwords expire after 60 days per CMS guidelines.

### Proposed Changes

- Add `devise-security` gem
- Styling for login page, password expired page, and password reset page
- Add tests for expired passwords, archived passwords, password resets, and user login

### Change Details

- Separated development and test databases in `database.yml`
- `devise-security` added devise errors in other languages
- Created `old_passwords` and `resources` tables for to keep track of expired passwords and archived passwords

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Screen Shot 2021-04-23 at 2 23 20 PM

Screen Shot 2021-04-23 at 2 23 39 PM

Screen Shot 2021-04-23 at 2 23 55 PM

Screen Shot 2021-04-23 at 2 24 11 PM

Screen Shot 2021-04-23 at 2 26 10 PM

### CODE REVIEW

File: dpc-impl/config/initializers/devise-security.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/config/initializers/devise-security.rb:1]: This is how the `devise-security` gem creates this file. Changing it to snake case will break it.

File: dpc-impl/db/schema.rb [hunk 2]
```diff
@@ -10,11 +10,25 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2021_04_09_181131) do
+ActiveRecord::Schema.define(version: 2021_04_23_155611) do
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/db/schema.rb:13]: Not for dates in the schema

File: dpc-impl/db/migrate/20210423155520_create_old_passwords.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateOldPasswords < ActiveRecord::Migration[6.1]
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/db/migrate/20210423155520_create_old_passwords.rb:1]: Not necessary for migration

File: dpc-impl/db/schema.rb [hunk 4]
```diff
@@ -10,11 +10,25 @@
...
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "old_passwords", force: :cascade do |t|
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/db/schema.rb:18]: Not necessary for migration

File: dpc-impl/db/schema.rb [hunk 5]
```diff
@@ -10,11 +10,25 @@
...
+    t.string "encrypted_password", null: false
+    t.string "password_archivable_type", null: false
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/db/schema.rb:20]: Not necessary for migration

**Review by dhgreene — APPROVED** [review]
👍

**Pull request merged**